### PR TITLE
[INFRA-1140] Make aladdin sso compatible

### PR DESCRIPTION
**Pull request by agianey**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -49,6 +50,12 @@ RUN pip install --upgrade pip && \
 
 # Update all needed tool versions here
 
+ARG AWS_CLI_VERSION=2.5.1
+RUN curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$AWS_CLI_VERSION.zip -o awscliv2.zip && \
+    unzip awscliv2.zip && \
+    ./aws/install && \
+    rm -rf aws && rm awscliv2.zip
```
Comment (agianey) [inline Dockerfile:57]: initially had this installed via poetry, but it didn't work well

File: aladdin-container.sh [hunk 2]
```diff
@@ -19,24 +19,6 @@ export PY_MAIN
 
 source "$SCRIPT_DIR/shared.sh"
 
-# Test user's aws configuration
```
Comment (agianey) [inline aladdin-container.sh:22]: I went back and forth on this, but I think it ended up being cleaner with aladdin being more unopinionated on this stuff

File: aladdin-container.sh [hunk 3]
```diff
@@ -118,8 +98,13 @@ function environment_init() {
 }
 
 function _is_cluster_ready() {
-    # Cluster is ready if we are on LOCAL or if we can pull kube config via kops
-    "$IS_LOCAL" || kops export kubecfg --name $CLUSTER_NAME --admin
+    # If not local, check if we can pull kube config via kops to determine if remote cluster is ready
+    if ! "$IS_LOCAL"; then
+        # Allow to use a different aws profile from aladdin config for kops command
```
Comment (agianey) [inline aladdin-container.sh:103]: Kops had an issue with its aws sdk being weird with sso, so it needed the option for a dedicated aws profile

File: aladdin-container.sh [hunk 4]
```diff
@@ -158,69 +143,6 @@ function _add_authentication_user_to_kubeconfig() {
 EOT
 }
 
-function _handle_aws_config() {
```
Comment (agianey) [inline aladdin-container.sh:161]: ripping out all the bastion stuff

File: aladdin.sh [hunk 5]
```diff
@@ -280,9 +280,7 @@ function enter_docker_container() {
         -e "HOST_ADDR=$HOST_ADDR" \
         -e "command=$command" \
         `# Mount host credentials` \
-        `# Mount destination for aws creds will not be /root/.aws because we will possibly make` \
-        `# changes there that we don't want propagated on the host's ~/.aws` \
-        -v "$(pathnorm ~/.aws):/root/tmp/.aws" \
+        -v "$(pathnorm ~/.aws):/root/.aws" \
```
Comment (agianey) [inline aladdin.sh:283]: no longer making changes to the ~/.aws files, so don't need to mount it to a unique location

File: pyproject.toml [hunk 6]
```diff
@@ -11,8 +11,7 @@ include = [
 
 [tool.poetry.dependencies]
 python = "^3.7"
-awscli = "1.20.28"
```
Comment (agianey) [inline pyproject.toml:14]: no longer using <v2

**Review by jarojasm95 — COMMENTED** [review]
looks good! left a small comment about `arm64` support

also lets update the version in `pyproject.toml`

**Review by sergey-korenets-fivestars — COMMENTED** [review]
LGTM

File: Dockerfile [hunk 7]
```diff
@@ -49,6 +50,12 @@ RUN pip install --upgrade pip && \
 
 # Update all needed tool versions here
 
+ARG AWS_CLI_VERSION=2.5.1
+RUN curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64-$AWS_CLI_VERSION.zip -o awscliv2.zip && \
```
[the remark this answers is not in the document]
Comment (agianey) [reply Dockerfile]: seems i had to use $(uname -m) to get x86_64 vs aarch64

**Review by jarojasm95 — APPROVED** [review]

**Pull request merged**